### ISSUE TITLE
Reject non-string and empty property name in `set()`

### ISSUE THREAD

**Issue by barmac** [bpmn-io/moddle#43]
__Describe the Bug__

Currently, when I pass a non-string or empty property name to `set()`, it's accepted by moddle. Non-strings are converted, so `NaN` becomes `"NaN"` and lands in $attrs. This allows buggy behavior like the one in https://github.com/bpmn-io/bpmn-js-properties-panel/issues/775

__Steps to Reproduce__

```javascript
import { Moddle } from 'moddle';

var cars = new Moddle([ carsJSON ]);

var taiga = cars.create('c:Car', { name: 'Taiga' });

taiga.set(undefined, 'hello');
taiga.set('', 'hello');
taiga.set(NaN, 'hello');

// no errors thrown
```

__Expected Behavior__

Moddle should throw a TypeError.

__Environment__

- Host (Browser/Node version), if applicable: [e.g. MS Edge 18, Chrome 69, Node 10 LTS]
- OS: [e.g. Windows 7]
- Library version: [e.g. 2.0.0]

**Comment by barmac** [comment]
I opened the issue to agree on how we want to solve the issue. I am willing to provide a solution after we decide.

**Comment by barmac** [comment]
Check out https://github.com/bpmn-io/moddle/pull/44

**Comment by nikku** [comment]
Released your fix as `v5.0.4`.
